assets/app/scripts/extensions/javaLink.js

```diff
@@ -1,86 +1,79 @@
 'use strict';
 
-/**
- * @ngdoc function
- * @name openshiftConsoleExtensions.extension:JavaLinkController
- * @description
- * # ProjectController
- * Controller of the openshiftConsole
- */
-angular.module('openshiftConsoleExtensions', ['openshiftConsole'])
-  .factory("ProxyPod", function(DataService) {
-    return function(namespace, podName, port) {
-      return new URI(DataService.url({
-        resource: 'pods/proxy',
-        // always use https when connecting to jolokia in a pod
-        name: ['https', podName, port || ''].join(':'),
-        namespace: namespace
-      }));
-    };
-  })
-  .run(function(ProxyPod, BaseHref, HawtioExtension, $templateCache, $compile, AuthService) {
+angular
+  .module('openshiftConsoleExtensions')
```

Comment:
@benjaminapetersen Still missing the call to `hawtioPluginLoader.addModule('openshiftConsoleExtensions')` in this branch.  Think it'd make more sense to make this extension a self-contained plugin with it's own name instead.  Also I'd personally wrap all the code in a self-executing function so I could do:

```
(function(angular, hawtioPluginLoader) {
  var pluginName = 'javaLinkExtension';
  angular.module(pluginName, [])
  // snip, then later...
  hawtioPluginLoader.addModule(pluginName);
})(angular, hawtioPluginLoader);